service/history/historyEngine.go

```diff
@@ -381,6 +381,28 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 		}
 	}
 
+	failoverMarkerInsertion := func(
+		shardNotificationVersion int64,
+		prevDomain *cache.DomainCacheEntry,
+		nextDomain *cache.DomainCacheEntry,
+	) error {
+
+		domainFailoverNotificationVersion := nextDomain.GetFailoverNotificationVersion()
+		domainActiveCluster := nextDomain.GetReplicationConfig().ActiveClusterName
+
+		if prevDomain != nil {
+			previousActiveCluster := prevDomain.GetReplicationConfig().ActiveClusterName
+			if prevDomain.IsGlobalDomain() &&
+				nextDomain.IsGlobalDomain() &&
+				domainFailoverNotificationVersion >= shardNotificationVersion &&
+				previousActiveCluster != domainActiveCluster &&
+				previousActiveCluster == e.currentClusterName {
+				return e.shard.InsertFailoverMarker(nextDomain.GetInfo().ID, nextDomain.GetFailoverVersion())
+			}
+		}
```

Comment:
if during the failover host encounters restart, the `prevDomain` will be nil since there is no prev value in cache.
if the above situation happens, there should be handling of this host handling over the domain to the to-be active host / cluster.

Reply:
Good point. So I am not going to use the previous domain entity. By checking the new domain entity failover end time should be able to tell if it should insert a marker

Reply:
Does the domain record contains failover from which cluster to which cluster?

Reply:
No. Currently, I don't store this data. What is the case to store a history of failovers